fxa-rust-client/ffi/src/ctypes.rs

```diff
@@ -16,3 +18,56 @@ impl From<SyncKeys> for SyncKeysC {
         }
     }
 }
+
+#[repr(C)]
+pub struct OAuthFlowInitC {
+    pub authorization_uri: *mut c_char,
+    pub flow_id: *mut c_char,
+}
+
+impl From<OAuthFlowInit> for OAuthFlowInitC {
+    fn from(init: OAuthFlowInit) -> Self {
+        OAuthFlowInitC {
+            authorization_uri: string_to_c_char(init.authorization_uri),
+            flow_id: string_to_c_char(init.flow_id),
+        }
+    }
+}
+
+#[repr(C)]
+pub struct OAuthInfoC {
+    pub access_token: *mut c_char,
+    pub keys_jwe: *mut c_char,
+    pub scopes: *mut c_char,
```

Comment:
Naming nit: if this is a single space-delimited string, I think we should use the singular "scope" rather than "scopes", which is in line with the terminology from the OAuth spec.  I like "scopes" when it's a list as it makes things more explicit.